pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -1335,10 +1339,13 @@ func TestUserSignupFailedToCreateDeactivationNotification(t *testing.T) {
 		AssertMetricsCounterEquals(t, 0, metrics.UserSignupUniqueTotal)
 
 		// A deactivated notification should not have been created
-		notification := &v1alpha1.Notification{}
-		err = r.client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Status.CompliantUsername + "-deactivated", Namespace: userSignup.Namespace}, notification)
-		require.Error(t, err)
-		require.Equal(t, "notifications.toolchain.dev.openshift.com \"john-doe-deactivated\" not found", err.Error())
+		labels := map[string]string{v1alpha1.NotificationUserNameLabelKey: userSignup.Status.CompliantUsername, v1alpha1.NotificationTypeLabelKey: "deactivated"}
+		opts := client.MatchingLabels(labels)
+
+		notificationList := &v1alpha1.NotificationList{}
+		err = r.client.List(context.TODO(), notificationList, opts)
```

Comment:
It's just a detail, but I would not use the labels in this test - in this way you also verify that it wasn't created without the labels:
```suggestion
		notificationList := &v1alpha1.NotificationList{}
		err = r.client.List(context.TODO(), notificationList)
```

Reply:
fixed here: eebffef30151239dd963cccf246d3c2dda86d070